pyproject.toml

```diff
@@ -1,10 +1,5 @@
 [build-system]
-requires = [
-    "cython~=3.0.2",
-    "oldest-supported-numpy",
-    "setuptools>=67.8",
-    "wheel",
-]
+requires = ["setuptools>=67.8", "cython~=3.0.2", "numpy>=1.25,<2"]
```

Comment:
Another question, seeing numpy included here: do you know for what reason fiona actually did have a build requirement on numpy before? A search in the codebase doesn't actually show any use of numpy (before this PR)

It seems this was brought in with https://github.com/Toblerity/Fiona/commit/37da85db8858a67a8a06451013af9aafd956ea33, which copied some code from rasterio (and rasterio indeed depends on numpy, but AFAIK Fiona does not?)